### DISCUSSION TITLE
Diode BAT46

### DISCUSSION THREAD

**Discussion by davyrio** [dzurikmiroslav/esp32-evse#25] [Q&A] [answered]
Hello Miroslav,

I'm a junior electronic. And, for my first real project i want to try to build my own EVSE. It's a challenge.
I looked many projects, but yours seems to be the best. It has inspired me for the electronics parts. 
For the software, i will write my own because it will be piloted by solar energy. 

After this long introduction, i come to my question. In your electronic schema all your BAT46 diode have the cathod oriented to the positive. Normaly shouldn't be to the opposite ?
I thank you for your help and congrats for your project.

**Comment by dzurikmiroslav** [comment, marked as answer]
Hi, thanks ;-)
It is ESP protection, you should google "esd suppressor diode" some like: https://blog.mbedded.ninja/electronics/circuit-design/esd-protection/protecting-io-lines-from-esd/

For charging from solar energy, I plan extend scripting api to communicate with solar inverter or energy meter through RS485/modbus. Thanks to that it will be possible power limiting charging...

**Reply by davyrio** [reply]
Wonderful ! i've all understood. Thanks

**Reply by davyrio** [reply]
Sorry @dzurikmiroslav, but i've an other question. Wich model of Socket TYPE2 had you choose for your EVSE. Most models don't have lock (solenoid or motor). Thank you.

**Reply by dzurikmiroslav** [reply]
I bought original Duosida Type2 socket with not probably original lock (no labels). Here is link https://www.evexpert.eu/p/339/interlocking-actuator-for-socket-type-2-female

**Reply by davyrio** [reply]
thanks.
